calcOutdatedMetrics.js

```diff
@@ -1,69 +1,80 @@
+const { exec } = require("child_process");
+
 process.stdin.resume();
-process.stdin.setEncoding('utf8');
-let data = '';
+process.stdin.setEncoding("utf8");
+let data = "";
 
-process.stdin.on('data', function (chunk) {
+process.stdin.on("data", function (chunk) {
   data += chunk;
 });
```

Comment:
Possible to refactor into methods?
Maybe something like `function dataSetup` etc, then have just one main fn kick off all the logic?

Reply:
Hmmm can you maybe describe more what this may look like?

The way I'm constructing `data` here is due to the fact that the `yarn outdated` output is being piped in, so it's a stream. If we refactor later to pass it in as an argument instead (as per other comment) we may be able to get rid of this.

Reply:
sure, kinda just refactoring into bite sized fns

Here's the gist of it (lol pun): https://gist.github.com/JonathanVanDyke/af9bca709b17f14d998a2126eb86d454
idk if you can open it tho since it's secret?
this all might be overkill if this is just in the early prototype stage tho, also got lost in the sauce a bit so it's prob not compilable code - just meant for reference

basically it kicks itself off with a self invoked fn up top to kind of hide the global scope a bit, in python I know they use something like `if __name__ === filename {do the things}` when it's a more simple single file script, so reaching for an equivalent here

Reply:
I think we still need `data` to be in the global scope. But I refactored a bit see what you think.